aggregation-worker/src/main/java/au/org/emii/geoserver/client/HttpIndexReader.java

```diff
@@ -0,0 +1,219 @@
+package au.org.emii.geoserver.client;
+
+
+import java.io.BufferedInputStream;
+import java.io.DataInputStream;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.*;
+
+import au.org.emii.aggregator.exception.AggregationException;
+import au.org.emii.download.DownloadRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+
+public class HttpIndexReader implements IndexReader {
+    private static final Logger logger = LoggerFactory.getLogger(HttpIndexReader.class);
+
+    protected String geoserver = null;
+
+    public HttpIndexReader(String geoserver) {
+        this.geoserver = geoserver;
+    }
+
+    @Override
+    public URIList getUriList(String profile, String timeField, String urlField, SubsetParameters subset) throws AggregationException {
```

Comment:
perhaps rename profile as layer - its what this actually is - profile refers to a very old gogoduck config mechanism.

Reply:
also probably just need the time parameters here.

Reply:
Time params are part of the SubsetParameters object.